libpromises/eval_context.c

```diff
@@ -1443,6 +1450,8 @@ static bool EvalContextClassPut(EvalContext *ctx, const char *ns, const char *na
         return false;
     }
 
+    StringSetAdd(ctx->all_classes, SafeStringDuplicate(name));
```

Comment:
No, didn't want to introduce anything custom so this is only class name.
I don't think storing a few hundred class names is that significant in terms or ram usage (in relation to entire application), but I agree it would be nice of there was of off switch.


For masterfiles there are around 400 classes set, 300 of them are set before we even check body agent where this is configured. So even it we keep it enabled until we check config and scrap&stop when we know it's off it would not save up much.